### ISSUE TITLE
Swagger 2.0 Schema Violations

### ISSUE THREAD

**Issue by dleguellecfortuitas** [bigcommerce/api-specs#582]
Catalog_v3.yml does not parse properly in https://editor.swagger.io/ nor via NSwagStudio
![Example](https://user-images.githubusercontent.com/38509772/148468069-7c329e0a-23d6-4fac-a772-8d280e08960c.PNG)

**Comment by el-besto** [comment]
A few other key collections for our integration project are broken outside of Stoplight:
- __Orders V3__ - violates with the use of `anyOf` on [line 2173](https://github.com/bigcommerce/api-specs/blob/7f99c30ba81369f53174b269c7993d1c86c1bdae/reference/orders.v3.yml#L2173) - perhaps upgrade to openapi 3.0.0?
- __Customers V2__ - needs to update its security definition [line 2900-2901](https://github.com/bigcommerce/api-specs/blob/7f99c30ba81369f53174b269c7993d1c86c1bdae/reference/customers.v2.yml#L2900)
- __Price List V2__ (see the violation error messages below)

## Customers V2 errors
```
Semantic error at security.1
Security requirements must match a security definition
Jump to line 2901
```

## Price List Errors
```
Structural error at info.contact.url
should match format "uri"
format: uri
Jump to line 83

Semantic error at paths./pricelists.get.parameters.8
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 146

Semantic error at paths./pricelists.get.parameters.9
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 149

Semantic error at paths./pricelists/{price_list_id}/records.get.parameters.16
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 903

Semantic error at paths./pricelists/{price_list_id}/records.get.parameters.17
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 906

Semantic error at paths./pricelists/assignments.get.parameters.4
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 2431

Semantic error at paths./pricelists/assignments.get.parameters.5
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 2435

Semantic error at paths./pricelists/assignments.get.parameters.6
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 2439

Semantic error at paths./pricelists/assignments.get.parameters.7
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 2443

Semantic error at paths./pricelists/assignments.delete.parameters.4
Schemas with 'type: array', require a sibling 'items: ' field
Jump to line 2471

Semantic error at security.0
Security requirements must match a security definition
Jump to line 4076
```

**Comment by slsriehl** [comment]
Thanks, @dleguellecfortuitas and @el-besto .  I've opened a ticket to review this.

**Comment by kamsar** [comment]
There are still validation errors in `catalog.v3.yml`, but they seem to be new compared to the OP's. The HTTP response types at the very end of the file appear to be written both as OpenAPI 3 and Swagger 2. Specifically the HTTP 207 response is also missing the Swagger 2 schema declaration, which breaks `openapi-typescript` creating types from it.

![image](https://user-images.githubusercontent.com/103677/154322757-59c0c5d7-2273-4022-b762-bbd9f3266f16.png)

Ex: `npx openapi-typescript https://raw.githubusercontent.com/bigcommerce/api-specs/master/reference/catalog.v3.yml --output catalog.ts`

**Comment by bc-andreadao** [comment]
I posted screenshots of updated schema violations in the ticket ([DEVDOCS-3442](https://jira.bigcommerce.com/browse/DEVDOCS-3442)). Because Stoplight supports both openapi 2.0 and 3.0, I did not correct schema violations that occurred in Swagger due to problems with openapi version. However, each schema violation is explained below and again in #628 

**Schema violations that were corrected** in #628 (i.e. problems that violated both openapi 2.0 and 3.0 formatting) include:
-Added "/" to basepath when path starts at host root, rather than leaving basepath empty (openapi 3.0 lets you leave the field empty to default to "/", but the term basepath implies 2.0 formatting)
-Added items field to array types. The items field includes they type for items in the array. This is required in openapi formatting.
-Fixed the formatting for multiple data types (adding oneOf before listing the 2 types). Simply listing the 2 types is valid JSON, but not openapi formatting.

**Schema violations in Swagger that did not need to be corrected in Stoplight**:
-"url" formatting is "uri" violation: Urls in openapi 2.0 have the uri format which does not accept empty strings. However, urls in openapi 3.0 have the uri-reference format, in which empty strings are accepted.
-anyOf is valid in openapi 3.0
-"no content as an additional property" violation: In openapi 3.0, responses can have content as an additional property

I opened new tickets to investigate changes for some schema violations that involved security requirements. These include:
-[DEVDOCS-3551](https://jira.bigcommerce.com/browse/DEVDOCS-3551)
-[DEVDOCS-3552](https://jira.bigcommerce.com/browse/DEVDOCS-3552)
